MAUI/DataForm/Validation.md

```diff
@@ -0,0 +1,224 @@
+
+You can validate the entire data form using the [ValidateForm] event of the data form.
+
+{% tabs %}
+{% highlight xaml %}
+
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:DataForm="clr-namespace:Syncfusion.Maui.DataForm;assembly=Syncfusion.Maui.DataForm"
+             x:Class="CommitMode.MainPage">
+
+    <DataForm:SfDataForm
+        x:Name="DataForm"
+        ValidateForm="OnDataFormValidateForm">
+    </DataForm:SfDataForm>
+
+</ContentPage>
+
+{% endhighlight %}
+{% highlight c# %}
+
+this.DataForm.ValidateForm += this.OnDataFormValidateForm;
+
+    private void OnDataFormValidateForm(object sender, DataFormValidateFormEventArgs e)
+    {
+        object dataObject = e.DataObject;
+        var errorMessage = e.ErrorMessage;
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Valid message
+
+If the value meets the desired criteria, you can show the [ValidMessage]. As error message, the valid message will also be displayed at the bottom of the editor.
+
+{% tabs %}
+{% highlight c# %}
+
+    [DataFormDisplayOptions(ColumnSpan = 2, ValidMessage = "Password strength is good")]
+    [Required(ErrorMessage = "Please enter the password")]
+    [RegularExpression(@"^(?=.*[a-z])(?=.*[A-Z])[a-zA-Z\d]{8,}$", ErrorMessage = "A minimum 8-character password should contain a combination of uppercase and lowercase letters.")]
+    public string Password { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+The [ValidMessageLabelTextStyle] property of [SfDataForm] and [DataFormItem] class is used to customize the text color, font, font size, font family and font  style of valid message label.
+The [ErrorLabelTextStyle] property of [SfDataForm] and [DataFormItem] class is used to customize the text color, font, font size, font family and font  style of error message label.
+
+N> The `ValidMessageLabelTextStyle` and `ErrorLabelTextStyle` property of `DataFormItem` class takes precedence when explicitly defined to particular editor otherwise where `SfDataForm.ErrorLabelTextStyle` and `SfDataForm.ErrorLabelTextStyle` will be applied to the editor.
+
+## Events
+
+### ValidateForm
+
+The [ValidateForm] event is used  to get the invalid DataFormItem’s details such as [DataObject], [NewValues] and [ErrorMessage].
+It occurs when [DataFormValidationMode] is `Manual`.
+
+{% endhighlight %}
+{% highlight c# %}
+
+this.DataForm.ValidateForm += this.OnDataFormValidateForm;
+
+    private void OnDataFormValidateForm(object sender, DataFormValidateFormEventArgs e)
+    {
+        object dataObject = e.DataObject;
+        var values = e.NewValues;
+        var errorMessage = e.ErrorMessage;
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+### ValidateProperty
+
+The [ValidateProperty] event is used to get the invalid DataFormItem’s details such as [IsValid], [PropertyName], [NewValue], [CurrentValue], [ValidMessage] and [ErrorMessage].
```

Comment:
The [ValidateProperty] event is used to get validation details of each DataFormItem such as [IsValid], [PropertyName], [NewValue], [CurrentValue], [ValidMessage] and [ErrorMessage].

You can also change validation status of a property by setting IsValid property and also change ErrorMessage and ValidationMessage

Reply:
content changed